src/cli/commands/publish.js

```diff
@@ -33,6 +35,19 @@ async function publish(
   if (access && access !== 'public' && access !== 'restricted') {
     throw new MessageError(config.reporter.lang('invalidAccess'));
   }
+  
+  commonDirectories.map((dir) => {
+    if (pkg[dir]) {
+      for (const file in pkg[dir]) {
+        fs.access(config.cwd + '/' + pkg[dir][file].replace('./', '')).then((res) => {
+          // valid files
+        }).catch((err) => {
+          config.reporter.error(`file not found: ${err.path}`);
+          process.exit(1);
+        });
```

Comment:
A few concerns:
- better use async/awaut than promisses
- process.exit does not seem like the right way to do, a command should not terminate process, we have a top level function for that in cli/index.js
- I don't understand the logic here: it checks that user has access to folders bin/main/modules/files folders? What does it verify?